lms/djangoapps/lti_provider/users.py

```diff
@@ -35,7 +35,10 @@ def authenticate_lti_user(request, lti_user_id, lti_consumer):
         )
     except LtiUser.DoesNotExist:
         # This is the first time that the user has been here. Create an account.
-        lti_user = create_lti_user(lti_user_id, lti_consumer)
+        lis_email = ""
+        if lti_consumer.auto_link_users_using_email:
+            lis_email = request.POST.get("lis_person_contact_email_primary", "")
+        lti_user = create_lti_user(lti_user_id, lti_consumer, lis_email)
```

Comment:
@Cup0fCoffee This is an interesting one. I have started using type hints in the new code I write that I have gotten into the habit of using empty strings instead of `None` to avoid the `str|None` in the function definitions. And as a boolean, it works out the same. That has made me use empty strings in more places than I probably should.

However, I think you are right, here the not sending that argument is a better way to communicate the absence of an email instead of an empty string.